oshi-core/src/main/java/oshi/util/Util.java

```diff
@@ -65,4 +71,35 @@ public static void sleepAfter(long startTime, long ms) {
             sleep(until - now);
         }
     }
+
+    /**
+     * Generates a licence key based upon the processor serial number, vendor,
+     * processor identifier, and total processor count. These are obtained using
+     * the interfaces like OperatingSystem, HardwareAbstractionLayer, CentralProcessor
+     * ComputerSystem and class SystemInfo.
+     *
+     */
+    static String generateLicenseKey()
+    {
+        SystemInfo systemInfo = new SystemInfo();
+        OperatingSystem operatingSystem = systemInfo.getOperatingSystem();
+        HardwareAbstractionLayer hardwareAbstractionLayer = systemInfo.getHardware();
+        CentralProcessor centralProcessor = hardwareAbstractionLayer.getProcessor();
+        ComputerSystem computerSystem = hardwareAbstractionLayer.getComputerSystem();
+
+        String vendor = operatingSystem.getManufacturer();
+        String processorSerialNumber = computerSystem.getSerialNumber();
+        String processorIdentifier = centralProcessor.getIdentifier();
+        int processors = centralProcessor.getLogicalProcessorCount();
+
+        String delimiter = "#";
```

Comment:
This is what the StackOverflow method used. Personally I would prefer hyphen-delimited hex strings using `String.format("%08x", foo.hashcode())` but that's optional.